SMTPServer/MailSession.cpp

```diff
@@ -56,85 +56,67 @@ int MailSession::SendResponse(int response_type)
 	char buf[RESPONSE_BUF_SIZE];
 	ZeroMemory(&buf, sizeof(buf));
 
-	if (response_type == Responses::WELCOME_TO_CLIENT)
-	{
-		strcpy(buf, "220 Welcome!\r\n");
-	}
-
-	else if (response_type == Responses::SERVICE_CLOSING)
-	{
-		strcpy(buf, "221 Service closing transmission channel\r\n");
-	}
-
-	else if (response_type == Responses::LOGIN_SUCCESS)
-	{
-		strcpy(buf, "235 Successfull login\r\n");
-	}
-
-	else if (response_type == Responses::OK)
-	{
-		strcpy(buf, "250 OK\r\n");
-	}
-
-	else if (response_type == Responses::LOGIN_RCV)
-	{
-		strcpy(buf, "334 Login has recieved\r\n");
-	}
-
-	else if (response_type == Responses::START_MAIL)
-	{
-		strcpy(buf, "354 Start mail input; end with <CRLF>.<CRLF>\r\n");
-	}
-
-	else if (response_type == Responses::SYNTAX_ERROR)
-	{
-		strcpy(buf, "501 Syntax error in parameters or arguments\r\n");
-	}
-
-	else if (response_type == Responses::COMMAND_NOT_IMPLEMENTED)
-	{
-		strcpy(buf, "502 Command not implemented\r\n");
-	}
-
-	else if (response_type == Responses::BAD_SEQUENSE)
-	{
-		strcpy(buf, "503 Bad sequence of commands\r\n");
-	}
-
-	else if (response_type == Responses::NO_USER)
-	{
-		strcpy(buf, "550 No such user\r\n");
-	}
-
-	else if (response_type == Responses::USER_NOT_LOCAL)
-	{
-		strcpy(buf, "551 User not local. Can not forward the mail\r\n");
-	}
-
-	else
-	{
-		strcpy(buf, "No description\r\n");
+	switch (response_type)
+	{
+	case Responses::WELCOME_TO_CLIENT:
+		strcpy(buf, WELCOME_STR);
+		break;
+	case Responses::SERVICE_CLOSING:
+		strcpy(buf, CLOSING_CHANNEL_STR);
+		break;
+	case Responses::LOGIN_SUCCESS:
+		strcpy(buf, LOGIN_STR);
+		break;
+	case Responses::OK:
+		strcpy(buf, OK_STR);
+		break;
+	case Responses::BAD_SEQUENSE:
+		strcpy(buf, BAD_SEQUENCE_STR);
+		break;
+	case Responses::COMMAND_NOT_IMPLEMENTED:
```

Comment:
PVS studio will warn about line 77 and 80, bacause case branches perform same action.
To avoid this warning, you just need write this two branches like this:
case Responses::COMMAND_NOT_IMPLEMENTED: case Responses::EMAIL_N_RECEIVED:
{...}